### ISSUE TITLE
Remove remnants of deferred checks from Bug 434570 et al

### ISSUE THREAD

**Issue by stephan-herrmann** [eclipse-jdt/eclipse.jdt.core#1074]
After #1077 the fix from https://bugs.eclipse.org/434570 seems to be unnecessary, as it is covered by the new fix. 

See https://bugs.eclipse.org/434570#c8 for an explanation that shows the similarity of both problems. Apparently, in #1077 we found a much more effective location for a similar fix.

Subsequent changes to be reverted
* parts of https://bugs.eclipse.org/434630
* all of https://bugs.eclipse.org/551426

**Comment by stephan-herrmann** [comment]
With the change of strategy from #1067 to #1077 also the focus of this polish ticket is moving.

**Comment by stephan-herrmann** [comment]
This was resolved via https://github.com/eclipse-jdt/eclipse.jdt.core/pull/1077/commits/4b898a2d4970d397c8e0870b91e731a55c2ddd26 in #1077
